hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java

```diff
@@ -184,6 +207,42 @@ public long getTotalLockTimeoutTransactionCount() {
     return totalLockTimeoutTransactionCount.value();
   }
 
+  public void setStartTimeOfLastIteration(long startTimeOfLastIteration) {
+    this.startTimeOfLastIteration.set(startTimeOfLastIteration);
+  }
+
+  public void setDurationOfLastIteration(long durationOfLastIteration) {
+    this.durationOfLastIteration.set(durationOfLastIteration);
+  }
+
+  public void setBlockChosenCountInLastIteration(
+      long blockChosenCountInLastIteration) {
+    this.blockChosenCountInLastIteration.set(blockChosenCountInLastIteration);
+  }
+
+  public void setContainerChosenCountInLastIteration(
+      long containerChosenCountInLastIteration) {
+    this.containerChosenCountInLastIteration.set(containerChosenCountInLastIteration);
+  }
+
+  public void resetLastIterationCounts() {
+    this.successCountInLastIteration.incr(-1L * this.successCountInLastIteration.value());
+    this.successBytesInLastIteration.incr(-1L * this.successBytesInLastIteration.value());
+    this.failureCountInLastIteration.incr(-1L * this.failureCountInLastIteration.value());
+  }
+
+  public void incrSuccessCountInLastIteration(long successCountInLastIteration) {
+    this.successCountInLastIteration.incr(successCountInLastIteration);
+  }
+
+  public void incrSuccessBytesInLastIteration(long successBytesInLastIteration) {
+    this.successBytesInLastIteration.incr(successBytesInLastIteration);
+  }
```

Comment:
Thanks @Tejaskriya for the patch.

Parameters can be renamed to `delta` to fix checkstyle failures:

```
hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/BlockDeletingServiceMetrics.java
 234: 'successCountInLastIteration' hides a field.
 238: 'successBytesInLastIteration' hides a field.
```

https://github.com/Tejaskriya/ozone/actions/runs/11813260671/job/32909976492